charts/public-service/templates/deployment.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "service.fullname" . }}
+  labels:
+    app: {{ template "service.name" . }}
+    chart: {{ template "service.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: {{ template "service.name" . }}
+      release: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+{{ toYaml .Values.rollingUpdateStrategy | indent 6 }}
+  minReadySeconds: 5
+  template:
+    metadata:
+      labels:
+        app: {{ template "service.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        ports:
+        - name: {{ .Values.portName }}
+          containerPort: {{ .Values.service.internalPort }}
+        # Inject (non sensitive) environment variables from the `Values` config.
+        env:
+        {{- range $key, $val := .Values.nonSensitiveEnvs }}
+        - name: {{ $key }}
+          value: {{ $val | quote }}
+        {{- end}}
+        # Inject (sensitive) environment variables from `Values` config, referring
+        # them from the related Secret object. This is an extra measure of security
+        # to avoid having credentials displayed in plain text in the template file.
+        # https://kubernetes.io/docs/concepts/configuration/secret/
+        {{- $chartName := include "service.fullname" . -}}
+        {{- range $key, $val := .Values.sensitiveEnvs }}
+        - name: {{ $key }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ $chartName }}-secret
+              key: {{ $key }}
+        {{- end}}
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: {{ .Values.portName }}
+          initialDelaySeconds: 15
+          periodSeconds: 10
+          timeoutSeconds: 5
+        readinessProbe:
```

Comment:
does it make sense to also make the `livenessProbe` and `readinessProbe` and their timeout, period, etc.. configurable? or we always for any public service gonna have them the same?

Reply:
https://github.com/commercetools/k8s-charts/pull/4#issuecomment-385906311